schematic/store/synapse.py

```diff
@@ -499,23 +506,24 @@ def login(
         Returns:
             synapseclient.Synapse: A Synapse object that is logged in
         """
-        # If no token is provided, try retrieving access token from environment
         if not access_token:
             access_token = os.getenv("SYNAPSE_ACCESS_TOKEN")
 
         # login using a token
         if access_token:
             try:
+                current_span = trace.get_current_span()
+                if current_span.is_recording() and access_token:
+                    user_info = info_from_bearer_auth(access_token)
+                    if user_info:
+                        current_span.set_attribute("user.id", user_info.get("sub"))
```

Comment:
We can't use the bearer token as the source for the user.id. this will not work when deployed to aws behind dca. Setting of the attribute must come from `syn.credentials.owner_id`

Reply:
Thanks, Bryan! I think I finally understand. Initially, I thought DCA would obtain a JWT token, decode it, and then pass the decoded token to Schematic. However, upon further thinking, it seems there’s no such "decoding" process. Instead, Schematic simply forwards the JWT token directly to Synapse.

Reply:
Schematic does decode the token and verifies the token against the public jwks of Synapse. This is how Synapse verifies that the token it was given was signed by Synapse and we can authenticate the authenticity of the request:
https://github.com/Sage-Bionetworks/schematic/blob/develop/schematic_api/api/security_controller.py#L17-L44

See this PR where I added the logic:
https://github.com/Sage-Bionetworks/schematic/pull/1493

The `sub` claim of the token however, does not contain the profile/owner_id of the actual user. Which is why it must get gotten after the login call when we get the user profile from Synapse.